src/ReachSets/discretize.jl

```diff
@@ -647,6 +682,45 @@ function discretize_interpolation(𝑆::InitialValueProblem{<:AbstractContinuous
     else
         throw(ArgumentError("input of type $(typeof(U)) is not allowed"))
     end
+    return Ω0, Ud
+end
 
-    return IVP(CLCDS(ϕ, Id(size(A, 1)), nothing, Ud), Ω0)
+# version using concrete operations with zonotopes
+function _discretize_interpolation_homog(X0, ϕ, Einit, set_operations::Val{:zonotope})
+    Einit = convert(Zonotope, Einit)
+    Z1 = convert(Zonotope, X0)
+    Z2 = linear_map(ϕ, minkowski_sum(X0, Einit))
+    Ω0 = overapproximate(ConvexHull(Z1, Z2), Zonotope)
+    return Ω0
+end
+
+# version using concrete operations with zonotopes
+function _discretize_interpolation_inhomog(δ, U0, U, X0, ϕ, Einit, Eψ0, A, sih, Phi2Aabs, set_operations::Val{:zonotope})
+    Einit = convert(Zonotope, Einit)
+    Eψ0 = convert(Zonotope, Eψ0)
+    Z1 = X0
+    ϕX0 = linear_map(ϕ, X0)
+    U0 = convert(Zonotope, U0)
+    δI = Matrix(δ*I, size(U0))
+    δU0 = linear_map(δI, U0)
+    Z2 = reduce(minkowski_sum, [ϕX0, δU0, Eψ0, Einit])
+    Ω0 = overapproximate(ConvexHull(Z1, Z2), Zonotope)
+
+    if U isa ConstantInput
+        Ud = ConstantInput(minkowski_sum(δU0, Eψ0))
+
+    elseif U isa VaryingInput
+        Ud = Vector{LazySet}(undef, length(U))
+        for (k, Uk) in enumerate(U)
+            Eψk = convert(Zonotope, sih(Phi2Aabs * sih(A * Uk)))
```

Comment:
Just a comment: I think it does not make sense to use a lazy `SymmetricIntervalHull` in this case because the whole set is computed.

Reply:
Ok. One could even do in the inner computation `linear_map(Phi2Aabs, linear_map(A, Uk))` to work with zonotopes. But i don't know if it has any benefit so i didn't add it.

Reply:
Good point, and I agree: since the set is immediately overapproximated with a box, this is probably not helpful.